src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -321,6 +320,35 @@ class LicenceService(
     )
   }
 
+  private fun deactivatePreviousLicenceVersion(licence: EntityLicence, username: String, fullName: String?) {
```

Comment:
presumably this doesn't need to call itself recursively as this is effectively ensuring that only a single previous active licence will exist?